man/mustache.5.ron

```diff
@@ -85,16 +92,113 @@ Output:
     * &lt;b&gt;GitHub&lt;/b&gt;
     * <b>GitHub</b>
 
+**Dotted Names**
+
+If the `name` contains dots, it is split on the dots to obtain multiple
+keys. The first key is looked up in the context as described above. If it
+is found, the next key is looked up within the previous result. This is
+repeated until a key is not found or until the last key is found. The
+final result is interpolated as above.
+
+Template:
+
+    * {{client.name}}
+    * {{age}}
+    * {{client.company.name}}
+    * {{{company.name}}}
+
+Hash:
+
+    {
+      "client": {
+        "name": "Chris & Friends",
+        "age": 50
+      },
+      "company": {
+        "name": "<b>GitHub</b>"
+      }
+    }
+
+Output:
+
+    * Chris &amp; Friends
+    *
+    *
+    * <b>GitHub</b>
+
+**Implicit Iterator**
+
+As a special case, if the `name` consists of only a dot and nothing else,
+the value that **is** the current context is interpolated as a whole. This
+is especially useful if the parent context is a list; see **Sections**
+below.
+
+Template:
+
+    * {{.}}
+
+Current context:
+
+    "Hello!"
+
+Output:
+
+    * Hello!
+
+**Lambdas**
+
+If any value found during the lookup is a callable object, such as a
+function or lambda, this object will be invoked with zero arguments. The
+value that is returned is then used instead of the callable object itself.
+
+An **optional** part of the specification states that if the final key in
```

Comment:
I am under the impression that lambdas *as a whole* are an optional part of the spec. Happy to be corrected if that's a misunderstanding.

Reply:
The `~lambdas` spec, as a whole, is optional. However, the non-optional `interpolation` and `sections` specs include lambda-related semantics as well:

https://github.com/mustache/spec/blob/7b09c52a149563d53156e7abcff3dde02f7b5b9d/specs/interpolation.yml#L12-L18

https://github.com/mustache/spec/blob/7b09c52a149563d53156e7abcff3dde02f7b5b9d/specs/sections.yml#L13-L22

I have interpreted this as meaning that an implementation that doesn't include the lambdas extension, should still take the return values of object methods rather than the methods themselves.

Reply:
That's a good catch, I stand corrected. I guess I never stumbled upon it stone the spec doesn't actually test that behavior outside of the lambda spec.